README.md

```diff
@@ -21,52 +21,58 @@ Java Rest Api Built with Spring Framework on a postgres database deployed to Her
 
 ## Endpoints
 
+All Endpoints that return lists take pageable query paramaters `page, size, & sort`
+
 #### User Routes
 
 | Method | Endpoint                | Access Control      | Description                                        |
 | ------ | ----------------------- | ------------------- | -------------------------------------------------- |
 | GET    | `/users/users`                | Admin               | Returns paginated list of all users                |
 | GET    | `/users/users/:userId`        | Admin               | Returns info for a single user.                    |
+| GET    | `/users/usertype`        | Admin               | Returns roles of current user.                    |
+| GET    | `/users/username/:username`        | Admin               | Returns a list of users with name starting with :username   |
 | POST   | `/users/newuser`              | Admin               | Creates a new user                                 |
 | PUT    | `/users/update-user/:userId`  | Admin               | Updates user with given id                         |
 | DELETE | `/users/users/:userId`        | Admin               | Deletes user with given id                         |
 
-#### Farmer Routes
+#### Farmer Routes - refers to client model with type "farmer"
 
 | Method | Endpoint                | Access Control | Description                                  |
 | ------ | ----------------------- | -------------- | -------------------------------------------- |
 | GET    | `/farmers/all`          | all users      | Returns a paginated list of all farmers      |
-| GET    | `/farmers/farmer/{id}`  | all users      | Returns Famer object with given ID           |
+| GET   | `/farmers/search`       | all users      | Returns paginated list of farmers with given criteria, takes query params `page, size, sort, name, location` |
+| GET    | `/farmers/farmer/{id}`  | all users      | Returns farmer object with given ID           |
 | POST   | `/farmers/add`  | all users      | Creates and returns a new farmer object |
-| POST   | `/farmers/search`       | all users      | Returns paginated list of farmers with given criteria |
 | PUT    | `/farmers/farmer/{id}`       | all users      | Modify and return updated farmer object with given id             |
 | DELETE | `/farmers/farmer/{id}`       | all users      | Delete a farmer with given id.                      |
 
 #### Organization Routes
```

Comment:
I would maybe put this below retailer so farmer and retailer are right next to each other, since they're so related